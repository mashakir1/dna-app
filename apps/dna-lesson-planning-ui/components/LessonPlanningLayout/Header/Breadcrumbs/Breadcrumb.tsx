// Imports
import useTranslation from "next-translate/useTranslation";

// Components
import { EmptyTick } from "components/Svg";

// Styles
import styles from "./breadcrumbs.module.scss";

interface BreadcrumbProps {
  title: string;
  active: boolean;
  completed: boolean;
  handleClick: () => void;
}

export const Breadcrumb = ({
  title,
  active,
  completed,
  handleClick,
}: BreadcrumbProps) => {
  const { t } = useTranslation("breadcrumbs");
  // TODO: HANDLE CLICK TO NAVIGATE
  // TODO - remove div as child of p
  return (
    <p
      className={`small ${styles.breadcrumb} ${active ? styles.active : ""} ${
        completed ? styles.completed : ""
      }`}
      aria-label={t(title)}
    >
      {completed && (
        <div className={styles.completedTick}>
          <EmptyTick colour="#30ccd6" fill="white" />
        </div>
      )}
      {t(title)}
    </p>
  );
};
