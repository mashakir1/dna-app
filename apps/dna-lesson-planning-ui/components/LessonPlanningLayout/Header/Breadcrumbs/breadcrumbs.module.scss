.breadcrumbs {
  list-style: none;
  overflow: hidden;
  border: 1px solid $colour_line_grey;
  width: fit-content;
  padding: 0;
  margin: 0;
  border-radius: 25px;
}

.breadcrumb {
  text-decoration: none;
  padding: $spacing-ks $spacing-xxl;
  background: white;
  position: relative;
  display: block;

  &::after {
    content: ' ';
    display: block;
    width: 0;
    height: 0;
    border-top: 25px solid transparent; /* Go big on the size, and let overflow hide */
    border-bottom: 25px solid transparent;
    border-left: 10px solid white;
    position: absolute;
    top: 50%;
    margin-top: -25px;
    left: 100%;
    z-index: 2;
  }

  &::before {
    content: ' ';
    display: block;
    width: 0;
    height: 0;
    border-top: 25px solid transparent;
    border-bottom: 25px solid transparent;
    border-left: 10px solid $colour_line_grey;
    position: absolute;
    top: 50%;
    margin-top: -25px;
    margin-left: 1px;
    left: 100%;
    z-index: 1;
  }

  &.active {
    background-color: $colour_light_teal;
    border-color: $colour_light_teal;

    &::before {
      border-left-color: white;
    }
    &::after {
      border-left-color: $colour_light_teal;
    }
  }
  &.completed {
    background-color: $colour_teal;
    border-color: $colour_teal;

    &::before {
      border-left-color: white;
    }

    &::after {
      border-left-color: $colour_teal;
    }

    .completedTick {
      position: absolute;
      left: 20px;
      border-radius: 25px;
      top: 50%;
      transform: translate(0, -45%);
    }
  }

  &:first-of-type {
    padding-left: $spacing-xl;

    &.completed {
      .completedTick {
        left: 10px;
      }
    }
  }
}
