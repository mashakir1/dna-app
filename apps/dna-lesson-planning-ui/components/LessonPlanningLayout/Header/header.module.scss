.header {
  width: 100vw;
  padding: $spacing-xxl;
  font-weight: 600;
  background-color: white;
}
