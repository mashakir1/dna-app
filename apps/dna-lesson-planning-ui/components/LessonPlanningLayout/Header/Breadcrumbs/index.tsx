import { Breadcrumb } from "./Breadcrumb";

import styles from "./breadcrumbs.module.scss";

interface BreadcrumbType {
  title: string;
  active: boolean;
  completed: boolean;
}

export const Breadcrumbs = ({
  breadcrumbs,
}: {
  breadcrumbs: Array<BreadcrumbType>;
}) => (
  <nav className={`${styles.breadcrumbs} flex`}>
    {breadcrumbs.map(({ title, active, completed }) => (
      <Breadcrumb
        key={title}
        title={title}
        active={active}
        completed={completed}
        // TODO:  handle click
        handleClick={() => {
          console.log("Clicked breadcrumb");
        }}
      />
    ))}
  </nav>
);
