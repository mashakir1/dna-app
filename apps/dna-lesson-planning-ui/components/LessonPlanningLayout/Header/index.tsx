// Imports
import useTranslation from "next-translate/useTranslation";

// Components
import { OutlineLinkButton } from "components/Buttons";
import { Breadcrumbs } from "./Breadcrumbs";

// Styles
import styles from "./header.module.scss";

const breadcrumbs = [
  {
    title: "choose-unit",
    active: true,
    completed: false
  },
  {
    title: "journey",
    active: false,
    completed: false
  },
  {
    title: "plan-lesson",
    active: false,
    completed: false
  },
  {
    title: "review",
    active: false,
    completed: false
  }
];

export const Header = () => {
  const { t } = useTranslation("common");
  return (
    <div className={`${styles.header} flex justify-between items-center`}>
      <Breadcrumbs breadcrumbs={breadcrumbs} />
      <OutlineLinkButton text={t("cancel")} href="/" />
    </div>
  );
};
