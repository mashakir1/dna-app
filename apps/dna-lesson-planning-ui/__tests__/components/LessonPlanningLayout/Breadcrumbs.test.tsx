import { render } from "@testing-library/react";

import { Breadcrumbs } from "components/LessonPlanningLayout/Header/Breadcrumbs";

const mockCrumbs = [
  {
    title: "choose-unit",
    active: true,
    completed: false
  },
  {
    title: "journey",
    active: false,
    completed: false
  },
  {
    title: "plan-lesson",
    active: false,
    completed: false
  },
  {
    title: "review",
    active: false,
    completed: false
  }
];

describe("Breadcrumbs", () => {
  it("should render the base breadcrumbs", () => {
    const { container } = render(<Breadcrumbs breadcrumbs={mockCrumbs} />);
    expect(container).toMatchSnapshot();
  });
  it("should render the updated breadcrumbs", () => {
    mockCrumbs[0].completed = true;
    mockCrumbs[1].active = true;
    const { container } = render(<Breadcrumbs breadcrumbs={mockCrumbs} />);
    expect(container).toMatchSnapshot();
  });
});
