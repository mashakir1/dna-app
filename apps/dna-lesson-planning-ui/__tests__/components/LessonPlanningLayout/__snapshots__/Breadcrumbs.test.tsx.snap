// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render the base breadcrumbs 1`] = `
<div>
  <nav
    class="breadcrumbs flex"
  >
    <p
      aria-label="choose-unit"
      class="small breadcrumb active "
    >
      choose-unit
    </p>
    <p
      aria-label="journey"
      class="small breadcrumb  "
    >
      journey
    </p>
    <p
      aria-label="plan-lesson"
      class="small breadcrumb  "
    >
      plan-lesson
    </p>
    <p
      aria-label="review"
      class="small breadcrumb  "
    >
      review
    </p>
  </nav>
</div>
`;

exports[`Breadcrumbs should render the updated breadcrumbs 1`] = `
<div>
  <nav
    class="breadcrumbs flex"
  >
    <p
      aria-label="choose-unit"
      class="small breadcrumb active completed"
    >
      <div
        class="completedTick"
      >
        <svg
          fill="none"
          height="30"
          viewBox="0 0 30 30"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="15"
            cy="15"
            fill="white"
            r="15"
          />
          <path
            clip-rule="evenodd"
            d="M22.5253 10.202C22.9158 10.5925 22.9158 11.2257 22.5253 11.6162L13.1503 20.9912C12.7598 21.3817 12.1266 21.3817 11.7361 20.9912L7.47472 16.7298C7.08419 16.3393 7.08419 15.7061 7.47472 15.3156C7.86524 14.9251 8.49841 14.9251 8.88893 15.3156L12.4432 18.8699L21.1111 10.202C21.5016 9.81146 22.1348 9.81146 22.5253 10.202Z"
            fill="#30ccd6"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      choose-unit
    </p>
    <p
      aria-label="journey"
      class="small breadcrumb active "
    >
      journey
    </p>
    <p
      aria-label="plan-lesson"
      class="small breadcrumb  "
    >
      plan-lesson
    </p>
    <p
      aria-label="review"
      class="small breadcrumb  "
    >
      review
    </p>
  </nav>
</div>
`;
